digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/ViiteTierekisteriClient.scala

```diff
@@ -182,14 +183,17 @@ object ViiteTierekisteriClient {
   }
 
   def sendJsonMessage(trProject:ChangeProject): ProjectChangeStatus ={
+    implicit val formats = DefaultFormats
     val request = new HttpPost(getRestEndPoint+"addresschange/")
     request.addHeader("X-Authorization", "Basic " + auth.getAuthInBase64)
     request.setEntity(createJsonmessage(trProject))
     val response = client.execute(request)
+    var errorMessage= TRErrorResponse("")
     try {
-    val statusCode = response.getStatusLine.getStatusCode
-    val reason = response.getStatusLine.getReasonPhrase
-    ProjectChangeStatus(trProject.id, statusCode, reason)
+      val statusCode = response.getStatusLine.getStatusCode
+      if (statusCode>=400 && statusCode<500)
+      errorMessage = parse(StreamInput(response.getEntity.getContent)).extract[TRErrorResponse] // would be nice if we didn't need case class for parsing of one attribute
```

Comment:
would it be possible to have something like parse(StreamInput(response.getEntity.getContent)).extractOpt[TRErrorResponse]..getOrElse(TRErrorResponse(""))
So we didnt need to declare errorMessage as var